book/src/dyn_tutorial/permissions.md

```diff
@@ -0,0 +1,126 @@
+| `leased`     | [Unique lease](./lease.md)         | the object is [leased](./lease.md) from another variable. This variable has unique access, but it is only temporary. The owner can cancel the lease and start using the object again.                                        |
+| `our leased` | [Shared lease](./shared_leases.md) | a [shared lease](./shared_leases.md) means that multiple variables have ([read-only](./sharing_xor_mutation.md)) access, but that access is only temporary. The owner can cancel the lease and start using the object again. |
+| `any`        |                                    | the variable can store values with any of the above permisions                                                                                                                                                               |
+
+## Shared ownership
+
+For example, instead of writing `my p`, we could've written `our p` to declare a variable with shared ownership:
+
+```
+class Point(x, y)
+
+async fn main() {
+    our p = Point(22, 44)
+}
+```
+
+
+## Enforcing uniqueness and ownership
+
+When you declare a variable as `my p`, that means that it has *unique ownership* of the object(s) stored inside of it. 
+
+```
+async fn main() {
+    our p = Point(22, 44)
+    my q = p                                         # Error
+}
+```
+
+
+
+## Permissions and mutation
+
+The most obvious effect 
+## Default permissions
+
+If you don't specify any ownership permision, the default is `our leased`. This means that the variable cannot be used to mutate the fields of the object. Example:
+
+```
+class Point(x, y)
+
+async fn main() {
+    p = Point(22, 44)      # declares `p` in `our leased` mode
+    p.x += 1               # Error!
+}
+```
+
+## Ownership permissions are enforced
+
+We saw that `my p` indicates that the variable `p` has unique ownership of its contents. But what happens if we try to store a value in `p` that cannot be owned? As an example, integers like `22` and `44` are inherently shared. If we try to claim ownership of those, we get an error. As an example, try running this:
+
+```
+async fn main() {
+    my x = 22               # Error!
+}
+```
+
+Now try changing the program to use `our x = 22`: you'll find that it works. You can also write `x = 22`, which is short for `our leased x = 22`. All of these work fine. This is because integers like `22` are shared: you cannot assert unique access to them. 
+
+**Exercise:** What do you think will happen with `leased x = 22` or `any x = 22`? Give it a try and see!
+
+## Ownership permissions on fields
+
+Ownership permissions also apply to fields and function arguments. The declaration of `Point` we have been using actually declares its fields with the default permissions, `our leased`:
```

Comment:
This would imply that structs with default-permissioned fields are tied to a stack discipline, and one would need to write all `my` and `our` fields to create a type that can be put on the heap without being tied to stack lifetimes?

Reply:
Yes, basically. My assumption is that `my` and `our` will be used all over the place on fields.

I'm not sure if this will feel good or not! I feel quite good about `my` and `our` on variables, and they are quite short, so I wanted to go for total consistency across all kinds of variables.

One other thought I've had is that the default for classes might be 'parametric', i.e., the owner of the class can 'raise up" the minimum level of ownership. But that's kind of the opposite of how modes typically work, where they "lower down" the full permissions (e.g., if you have `class Pair(my a, my b)` and then you have a `our pair: Pair`, you only have shared access to `pair.a` and `pair.b`).

Reply:
Starting with consistency is best.